lib/query.js

```diff
@@ -93,9 +84,7 @@ function eachPage(pageCallback, done) {
                     next = inner;
                 } else {
                     next = function() {
-                        if (done) {
-                            done(null);
-                        }
+                        done(null);
```

Comment:
The check on line 68 and corresponding error message on line 69 says it's valid for `done` to be `undefined`, which would make this line crash. Is that not possible because of `callbackToPromise`? If so, we should update the check on line 68-69

Reply:
The end use of `eachPage` can allow `done` to be `undefined` and in that case it will be defined by `this.eachPage = callbackToPromise(eachPage, this, 1);` on query line 30. So it is an error message that make sense from an end user perspective.

Reply:
Ah gotcha, thanks for the explanation